UniqueArrayImp.h

```diff
@@ -0,0 +1,116 @@
+#ifndef MTMPARKINGLOT_UNIQUEARRAYIMP_H
+#define MTMPARKINGLOT_UNIQUEARRAYIMP_H
+
+template <class Element, class Compare>
+UniqueArray<Element, Compare>::UniqueArray(unsigned int size) :
```

Comment:
Maybe I'm misunderstanding, but what if the user wishes to have a different (other the default) comparison function?